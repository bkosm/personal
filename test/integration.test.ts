import { assertTitle, intTest } from "./deps.ts";
import { assertVisualSnapshot } from "./regression.ts";
import { assertSnapshot } from "https://deno.land/std@0.151.0/testing/snapshot.ts";
import { delay } from "https://deno.land/std@0.150.0/async/delay.ts";

intTest("webdriver tests", async (t, b) => {
  const p = await b.newPage();
  p.setViewport({ width: 1920, height: 1080 });

  // deno-lint-ignore no-explicit-any
  const open = async (page: any, path: any) =>
    await page.goto(`http://localhost:8000${path}`, {
      waitUntil: "networkidle2",
    });

  // Index page
  await open(p, "/");

  await t.step("index page - has proper title", async () => {
    await assertTitle(t, p);
  });

  await t.step("index page - visual requirements are met", async () => {
    await assertVisualSnapshot(p, "index1");
  });

  // Post page
  await open(p, "/posts/test-page");

  await t.step("post page - visual requirements are met", async () => {
    await delay(100);
    await assertVisualSnapshot(p, "posts1");
  });

  // Invalid page
  await open(p, "/posts/whoops-not-gonna-happen");

  await t.step("invalid post page - redirects home", () => {
    assertSnapshot(t, p.url());
  });
});
