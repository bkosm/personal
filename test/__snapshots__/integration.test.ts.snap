export const snapshot = {};

snapshot[`page title 1`] = `"bkosm - fresh blog"`;

snapshot[`page title 2`] = `"Test page used for automated testing - bkosm"`;

snapshot[`post url 1`] = `"http://localhost:8000/posts/test"`;

snapshot[`redirect url 1`] = `"http://localhost:8000/"`;
